homework/src/index.js

```diff
@@ -31,17 +31,66 @@
   }
 
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
+    const root = document.getElementById("root");
+    const div = createAndAppend("div", root, { class: "container" });
+    const header = createAndAppend("p", div, {
+      text: "HYF Repositories",
+      class: "header"
+    });
+    const table = createAndAppend("table", root, { class: "table" });
+    const select = createAndAppend("select", header, { class: "select" });
+
+    const tbody = createAndAppend("tbody", table);
+    const tr = createAndAppend("tr", tbody);
+
+    createAndAppend("td", tr, tbody, { text: "Repository: ", class: "label" });
+    const repository = createAndAppend("a", tbody);
+    const description = createAndAppend("tr", tbody);
+    const forks = createAndAppend("tr", tbody, { class: "label" });
+
+    const updated = createAndAppend("tr", tbody, { class: "label" });
+
+    fetchJSON(url, (error, repositories) => {
+      if (error) {
+        createAndAppend("div", root, {
+          text: error.message,
+          class: "alert-error"
+        });
       } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
       }
+
+      repositories.forEach((element, index) => {
+        createAndAppend("option", select, {
+          text: element.name,
+          value: index
+        });
+      });
+
+      function repositoryBox() {
+        const index = select.value;
+        repository.innerText = select[index].textContent;
+        repository.setAttribute("href", repositories[index].html_url);
+        repository.setAttribute("target", "_blank");
+
+        forks.innerHTML = "Forks: " + repositories[index].forks;
+        description.innerHTML =
+          "Description: " + repositories[index].description;
+        updated.innerHTML =
+          "Updated: " +
+          new Date(repositories[index].updated_at).toLocaleString();
+      }
+
+      repositoryBox();
```

Comment:
There is no point in defining an inline function and then immediately calling as you are doing here. You could easily comment out lines 70, 82 and 84 without it changing anything.

It is a good idea however to create a function and place it at file level, so not inside the event handler function. But then, you must pass all the information that that function requires through the parameter list. If you do that your code becomes better readable, because your function `repositoryBox` has no longer any external dependencies. The resulting code could look like this (leaving aside other coding issues that still remain in this code):

```js

    repositoryBox(repositories[0], repositoryLink, forks, description, updated);

    select.addEventListener("change", () => {
      repositoryBox(repositories[select.value], repositoryLink, forks, description, updated);
    });
  });
}

function repositoryBox(repository, repositoryLink, forks, description, updated) {
  repositoryLink.innerText = repository.name;
  repositoryLink.setAttribute("href", repository.html_url);
  repositoryLink.setAttribute("target", "_blank");

  forks.innerHTML = "Forks: " + repository.forks;
  description.innerHTML =
    "Description: " + repository.description;
  updated.innerHTML =
    "Updated: " +
    new Date(repository.updated_at).toLocaleString();
}
```

Furthermore, function names should contain a verb in their names as they perform an action (do this, do that, etc.), e.g. `renderRepositoryBox`.